downstream.go

```diff
@@ -256,6 +261,13 @@ func (dc *downstreamConn) readMessages(ch chan<- event) error {
 	return nil
 }
 
+func (dc *downstreamConn) flushIdleMessages() {
+	for _, m := range dc.idleMessages {
+		dc.conn.SendMessage(m)
+	}
+	dc.idleMessages = make([]*irc.Message, 0, dc.srv.IdleLimit)
```

Comment:
Instead of re-allocating the slice, we can probably do something like this to set its length to zero:

```go
dc.idleMessages = dc.idleMessages[:0]
```